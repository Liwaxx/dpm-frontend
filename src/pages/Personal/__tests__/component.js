import React from 'react';
import renderer from 'react-test-renderer';
import Component from '../component';

jest.mock('../../../components/elements/AddSchedule');
jest.mock('../../../components/elements/Calendar');
jest.mock('../../../components/elements/Card');
jest.mock('../../../components/elements/Detail');
jest.mock('../../../components/elements/Dropdown');
jest.mock('../../../components/elements/GridHeatmap1');
jest.mock('../../../components/elements/GridHeatmap2');
jest.mock('../../../components/elements/ListSchedule');
jest.mock('../../../components/elements/Modal');
jest.mock('../../../components/elements/ModalPertahun');
jest.mock('../../../components/elements/ProductCard');
jest.mock('../../../components/elements/PersonalScore');
jest.mock('../../../components/elements/PieChart');
jest.mock('../../../components/elements/Text');
jest.mock('../../../utils/dateFormat');

describe('Personal', () => {
  let props = {};
  beforeEach(() => {
    props = {
      actions: {
        fetchDataProduct: jest.fn(),
        fetchDataNotif: jest.fn(),
        fetchPersonalScore: jest.fn(),
        fetchSquadBacklog: jest.fn(),
        fetchPersonalBacklog: jest.fn(),
        fetchWorkIntensity: jest.fn(),
        fetchListSchedule: jest.fn(),
        getDetailP: jest.fn(),
        getDetailS: jest.fn(),
        fetchDataTimelinSkill: jest.fn(),
        fetchDataProductReport: jest.fn(),
      },
      classes: {},
      isLoading: false,
      dataWorkIntensity: {},
      addSchedule: {},
      listSchedule: {},
      dataProduct: {},
      notifications: {},
      dataScore: {},
      dataSB: {},
      dataPB: {},
      dataDetailS: {},
      dataDetailP: {},
    };
  });

  it('renders correctly', () => {
    const tree = renderer
      .create(<Component {...props} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});