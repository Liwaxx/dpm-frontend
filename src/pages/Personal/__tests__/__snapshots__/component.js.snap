// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Personal renders correctly 1`] = `
<section>
  <div>
    <div />
    <div>
      <div />
      <div />
    </div>
    <div />
  </div>
  <div>
    <div />
    <div />
  </div>
  <div>
    <div>
      <div
        className="MuiPaper-root-1 MuiPaper-elevation2-5 MuiPaper-rounded-2"
        style={
          Object {
            "display": "none",
          }
        }
      />
      <div
        className="MuiPaper-root-1 MuiPaper-elevation2-5 MuiPaper-rounded-2"
        style={
          Object {
            "display": "block",
          }
        }
      />
    </div>
  </div>
  <div />
</section>
`;
